crates/ruff/src/rules/ruff/mod.rs

```diff
@@ -1,6 +1,7 @@
 //! Ruff-specific rules.
 
 pub(crate) mod rules;
+pub(crate) mod typing;
```

Comment:
Is this the correct place to keep the common functionality? We could move this to `ruff_python_ast` crate which would add dependencies to other crates mainly `ruff_python_semantic`.

Reply:
It's probably okay for now, but you may be able to make it `pub(super)`.